### PR TITLE
Fix islock attribute

### PR DESCRIPTION
**Pull request by Cofresi**
With v0.14 simple transactions are a thing of the past once spork 20 is active. That means all tx should benefit from automatic instant send. The corresponding logic to set the attribute `txlock` can be simplified accordingly.

Also updated several dependencies.

### CODE REVIEW

File: lib/services/dashd.js [hunk 1]
```diff
@@ -2248,16 +2248,8 @@ Dash.prototype.getDetailedTransaction = function(txid, callback) {
...
-    if (tx.feeSatoshis >= Dash.DEFAULT_INSTANTSEND_FEE) { // if transaction fee is >= 0.0001 check for txlock
-      var rawTx = new dashcore.Transaction(result.hex);
-      var hash = dashcore.crypto.Hash.sha256sha256(rawTx.toBuffer());
-      var id = hash.toString('binary');
-      if (self.node.services.dashd.zmqKnownTransactionLocks.get(id)) {
-        tx.txlock = true; // if "txlock" notification received through ZMQ, set txlock = true
-      }
-    }
+  function addTxlockToTx(tx, result) {
+    tx.txlock = result.instantlock || result.chainlock;
```
Comment (antouhou) [inline lib/services/dashd.js:2252]: But chainlock isn't the same thing as IS lock, isn't it? IS lock prevents double spend within one block, chainlock just blocks possibility of the reorgs after the block was mined

**Review by antouhou — CHANGES_REQUESTED** [review]

**Review by shumkov — DISMISSED** [review]
👍 utACK

File: lib/services/dashd.js (hunk 1 above)
Comment (shumkov) [reply lib/services/dashd.js:2252]: I think from a user perspective it's kinda the same - a transaction is confirmed.

**Comment by codablock** [comment]
I'm wondering if after this change Insight will be able to indicate IS locks when the lock came after the TX, while the first request for TX did not have the lock yet. Is there internal caching involved between Insight and the RPC calls? The old solution looked into what ZMQ gave you, so it was ensured to be up-to-date...but if in this solution the RPC calls are cached (that's my assumption here), you'll never get to know about the IS lock if its already in the cache.

File: lib/services/dashd.js (hunk 1 above)
Comment (Cofresi) [reply lib/services/dashd.js:2252]: @antouhou please check the relevant exchange in 14-0-dev where this topic was discussed. Esentially with a chainlock the instantlock field is set to true as well. So there will never be a situation where `.chainlock` is true and `.instantlock` is false.
https://dashpay.slack.com/archives/G7W56UMCG/p1553618255080300?thread_ts=1553578894.065300&cid=G7W56UMCG

**Comment by Cofresi** [comment]
> I'm wondering if after this change Insight will be able to indicate IS locks when the lock came after the TX, while the first request for TX did not have the lock yet. Is there internal caching involved between Insight and the RPC calls? The old solution looked into what ZMQ gave you, so it was ensured to be up-to-date...but if in this solution the RPC calls are cached (that's my assumption here), you'll never get to know about the IS lock if its already in the cache.

Good point. There is a cache involved for transactions, but it is only 30 sec. I did think about this before I removed the zmq cache check and based my action on the following reasoning: most of the time this shouldn't be an issue as the direct verification of a tx would not happen in the short window of time between the 1st arrival of the tx (with `.txlock = false`) and the arrival of the ISLOCK (which sets`.txlock = true`). There is a case which I didn't think about though. Some people like to keep an address window open to be able see incoming tx in real time. That page tries to embed the new tx by calling the tx endpoint for that new tx with `/insight-api/tx/<newTxID>`. This will happen immediately and in most cases put the tx in the cache with `.txlock = false`. However, there is no info shown about the txlock status of this new tx on the page. And in 30 sec the cache is released and any new call to `/insight-api/tx/<newTxID>` will show the correct information with `"txlock":true`. Because of this I don't really see any added value in having this check of the zmq message cache.

**Review by antouhou — APPROVED** [review]
utACK

**Pull request merged**